Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.mm

```diff
@@ -296,24 +321,22 @@
     return status != noErr && status != kVTVideoDecoderReferenceMissingErr;
 }
 
-void WebCoreDecompressionSession::handleDecompressionOutput(bool displaying, OSStatus status, VTDecodeInfoFlags, CVImageBufferRef rawImageBuffer, CMTime presentationTimeStamp, CMTime presentationDuration)
+Expected<RetainPtr<CMSampleBufferRef>, OSStatus> WebCoreDecompressionSession::handleDecompressionOutput(bool displaying, OSStatus status, VTDecodeInfoFlags, CVImageBufferRef rawImageBuffer, CMTime presentationTimeStamp, CMTime presentationDuration)
 {
     assertIsCurrent(m_decompressionQueue.get());
 
     if (isNonRecoverableError(status)) {
         RELEASE_LOG_ERROR(Media, "Video sample decompression failed with error:%d", int(status));
-        m_lastDecodingError = status;
```

Comment:
Does `handleDecompressionOutput` needs to be a method, can it be a free function?